python/semantic_kernel/connectors/ai/chat_completion_client_base.py

```diff
@@ -62,7 +68,28 @@ def _prepare_chat_history_for_request(
         """
         Prepare the chat history for a request, allowing customization of the key names for role/author,
         and optionally overriding the role.
+
+        ChatRole.TOOL messages need to be formatted different than system/user/assistant messages:
+            They require a "tool_call_id" and (function) "name" key, and the "metadata" key should
+            be removed. The "encoding" key should also be removed.
+
+        Arguments:
+            chat_history {ChatHistory} -- The chat history to prepare.
+
+        Returns:
+            List[Dict[str, Optional[str]]] -- The prepared chat history.
         """
-        return [
-            message.model_dump(exclude_none=True, exclude=["metadata", "encoding"]) for message in chat_history.messages
-        ]
+        prepared_messages = []
+        for message in chat_history.messages:
+            if message.role == ChatRole.TOOL:
+                dump = message.model_dump(exclude_none=True, exclude=["encoding"])
+                if "metadata" in dump and "tool_call_id" in dump["metadata"]:
```

Comment:
the specifics of openaichatmessage should probably be in there, have a look at the way I did it in my chat history xml PR, solved it slightly differently (although I forgot the function_name)